### PR TITLE
Reset debug views on teardown

### PR DESCRIPTION
**Pull request by FrancescoC-unity**
As in the title, this also fixes the issue with the sky and matcap left on when closing

### CODE REVIEW

File: com.unity.render-pipelines.core/Runtime/Utilities/CoreUtils.cs [hunk 1]
```diff
@@ -712,6 +712,15 @@ public static bool IsSceneLightingDisabled(Camera camera)
             return disabled;
         }
 
+        public static void ForceSceneLightingEnable()
+        {
+            // Determine whether the "No Scene Lighting" checkbox is checked for the current view.
```
Comment (FrancescoC-unity) [inline com.unity.render-pipelines.core/Runtime/Utilities/CoreUtils.cs]: Copy-pasted comment, let me remove it

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipeline.cs [hunk 2]
```diff
@@ -826,6 +826,11 @@ void DisposeProbeCameraPool()
             }
 
             CameraCaptureBridge.enabled = false;
+
+            // Disable all debug views
+            DebugManager.instance.Reset();
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipeline.cs]: I am wondering if this is ok? I guess we create and delete render pipeline various time already. Have you try for example to enable the matcap view, then change a parameter on HDRP asset? Do it diasble the matcap view?

Comment (FrancescoC-unity) [reply com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipeline.cs]: Uhm you are right, haven't thought of it;  it actually leaves stuff in a weird state, investigating.

Comment (FrancescoC-unity) [reply com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipeline.cs]: So, it disables the matcap on our side, however, the engine reverts for some reason to the old no scene lighting debug view...

**Comment by FrancescoC-unity** [comment]
Yeah this is going to give some grievances when swapping asset due to something on C++ side. Closing this, I am going to fix the matcap sky issue in another way.

**Pull request closed without merging**